src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -757,22 +765,49 @@ impl FlowServiceImpl {
                             matches!(compaction_rule, CompactionRule::MetadataOnly(_));
                     };
 
-                    LogicalPlan::HardCompactionDataset(HardCompactionDataset {
+                    Ok(LogicalPlan::HardCompactionDataset(HardCompactionDataset {
                         dataset_id: flow_key.dataset_id.clone(),
                         max_slice_size,
                         max_slice_records,
                         keep_metadata_only,
-                    })
+                    }))
+                }
+                DatasetFlowType::Reset => {
+                    let new_head_hash = if let Some(config_rule) = config_snapshot
+                        && let FlowConfigurationSnapshot::Reset(reset_rule) = config_rule
+                    {
+                        Some(reset_rule.new_head_hash.clone())
```

Comment:
**Q:** What if I, as an API user, specify as a new head a hash that is already a head for a dataset?

maybe in that case, we could not perform the action chain (flow -> task -> write files), but immediately reply to the user that "hash is already a head"?

Reply:
Done